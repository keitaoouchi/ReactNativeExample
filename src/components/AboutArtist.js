import React from "react";
import {ScrollView} from "react-native";
import {Actions} from "react-native-router-flux";
import ArtistHeader from "./ArtistHeader";
import Workaround from "../misc/Workaround";
import HorizontalAlbumCoverCollection from "./HorizontalAlbumCoverCollection";
import RelatedArtistCollection from "./RelatedArtistCollection";
import Color from "../misc/Color";
import AboutArtistActionCreator from "../flux/action/AboutArtistActionCreator";
import AboutArtistStore from "../flux/store/AboutArtistStore";
import {Dispatcher} from "flux";

const AboutArtistPresentation = ({artist, albums, singles, relatedArtists}) => (
  <ScrollView
    horizontal={false}
    style={{
      flex: 1,
      backgroundColor: Color.black, ...Workaround.style.root
    }}
    showsVerticalScrollIndicator={true}
    alwaysBounceVertical={true}
    bounces={true}
    directionalLockEnabled={true}
    automaticallyAdjustContentInsets={false}
  >
    <ArtistHeader
      artist={artist}
    />
    <HorizontalAlbumCoverCollection
      title="アルバム"
      albums={albums}
      onPress={(album) => Actions.album({title: album.title, album, hideNavBar: false})}
    />
    <HorizontalAlbumCoverCollection
      title="シングル"
      albums={singles}
      onPress={(album) => Actions.album({title: album.title, album, hideNavBar: false})}
    />
    <RelatedArtistCollection
      title="関連アーティスト"
      artists={relatedArtists}
      onPress={(artist) => Actions.artist({title: artist.name, artist, hideNavBar: false})}
    />
  </ScrollView>
);

export default class AboutArtist extends React.Component {

  constructor(props) {
    super(props);

    const dispatcher = new Dispatcher();
    this.actionCreator = new AboutArtistActionCreator(dispatcher);
    this.store = new AboutArtistStore(dispatcher);
  }

  componentDidMount() {
    this.store.addListener(() => this.forceUpdate());
    this.actionCreator.reloadArtist(this.props.artist);
    this.actionCreator.fetchAlbums(this.props.artist);
    this.actionCreator.fetchRelatedArtists(this.props.artist);
  }

  render() {
    const state = this.store.getState();
    return <AboutArtistPresentation
      artist={state.artist || this.props.artist}
      albums={state.albums}
      singles={state.singles}
      relatedArtists={state.relatedArtists}
    />;
  }
}
