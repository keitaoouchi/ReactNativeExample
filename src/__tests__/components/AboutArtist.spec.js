import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import AboutArtist from "../../../src/components/AboutArtist";
import ArtistFactory from "../factories/ArtistFactory";
import FetchMock from "fetch-mock";

describe('AboutArtist', () => {

  beforeAll(() => {
    FetchMock.get('*', {});
  });

  it('renders correctly', () => {
    const artist = ArtistFactory.build('Artist');
    const tree = renderer.create(
      <AboutArtist artist={artist}/>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});