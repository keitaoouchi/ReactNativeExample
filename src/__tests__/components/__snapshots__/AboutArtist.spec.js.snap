// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutArtist renders correctly 1`] = `
<RCTScrollView
  alwaysBounceVertical={true}
  automaticallyAdjustContentInsets={false}
  bounces={true}
  directionalLockEnabled={true}
  horizontal={false}
  showsVerticalScrollIndicator={true}
  style={
    Object {
      "backgroundColor": "#181818",
      "flex": 1,
      "marginTop": 64,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "height": 300,
          "justifyContent": "flex-end",
          "width": 320,
        }
      }
    >
      <Image
        onLoadEnd={[Function]}
        source={
          Object {
            "uri": "https://example.com/1",
          }
        }
        style={
          Object {
            "borderRadius": 80,
            "height": 160,
            "opacity": 0,
            "width": 160,
          }
        }
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 20,
            "margin": 16,
          }
        }
      >
        Name-1
      </Text>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#1ed760",
                "borderRadius": 22,
                "height": 44,
                "margin": 12,
                "padding": 8,
                "width": 200,
              },
              false,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "white",
                "fontSize": 16,
                "fontWeight": "bold",
                "margin": 4,
                "textAlign": "center",
              }
            }
          >
            Spotifyで聴く
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "marginBottom": 8,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 16,
            "margin": 8,
          }
        }
      >
        アルバム
      </Text>
      <RCTScrollView
        alwaysBounceVertical={false}
        automaticallyAdjustContentInsets={false}
        bounces={true}
        directionalLockEnabled={true}
        horizontal={true}
        showsVerticalScrollIndicator={false}
        style={
          Object {
            "height": 140,
          }
        }
      >
        <View />
      </RCTScrollView>
    </View>
    <View
      style={
        Object {
          "marginBottom": 8,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 16,
            "margin": 8,
          }
        }
      >
        シングル
      </Text>
      <RCTScrollView
        alwaysBounceVertical={false}
        automaticallyAdjustContentInsets={false}
        bounces={true}
        directionalLockEnabled={true}
        horizontal={true}
        showsVerticalScrollIndicator={false}
        style={
          Object {
            "height": 140,
          }
        }
      >
        <View />
      </RCTScrollView>
    </View>
    <View
      style={
        Object {
          "marginBottom": 8,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 16,
            "margin": 8,
          }
        }
      >
        関連アーティスト
      </Text>
      <RCTScrollView
        alwaysBounceVertical={false}
        automaticallyAdjustContentInsets={false}
        bounces={true}
        directionalLockEnabled={true}
        horizontal={true}
        showsVerticalScrollIndicator={false}
        style={
          Object {
            "height": 180,
          }
        }
      >
        <View />
      </RCTScrollView>
    </View>
  </View>
</RCTScrollView>
`;
